### PR TITLE
Return message when database does not exists

### LINKED ISSUE #341: Suite status endpoint does not return proper message when DB does not exist
### Version used

* `AET 2.1.6`

### Context of the issue/feature
When running new suite and DB for it does not exist and auto creation of DBs is disabled - there is no message what happened but suite run fails.

### Steps to reproduce

1. Make sure that `MongoDBClient` `allowAutoCreate` is set to `false` (disable auto creation of DBs in AET).
2. Run suite (e.g. using `aet.sh`) with `company` and `project` that does not have DB created.

Result:
Suite is not processed and no progress is presented by suite status endpoint:
```json
{"status":"UNKNOWN","message":""}
```
From the user (who runs `aet.sh`) perspective, nothing happens. Internally AET does not process the suite.

Expected:
[Suite status endpoint](https://github.com/Cognifide/aet/blob/master/test-executor/src/main/java/com/cognifide/aet/executor/SuiteStatusServlet.java) returns proper status (e.g. `PROCESSING_ERROR`) and message about the problem.

### Appendix
This case is already supported by [`MetadataDAOMongoDBImpl`](https://github.com/Cognifide/aet/blob/master/core/datastorage/src/main/java/com/cognifide/aet/vs/metadata/MetadataDAOMongoDBImpl.java#L184). It looks that this exception is not handled properly.

### PR DESCRIPTION
**Pull request by Asia95**
When database for given company and project doesn't exists proper message is displayed.

## Description
[SuiteStausServlet ](https://github.com/Cognifide/aet/blob/master/test-executor/src/main/java/com/cognifide/aet/executor/SuiteStatusServlet.java) returns status `PROCESSING_ERROR` and proper message when database doesn't exists

## Motivation and Context
Closes #341 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] I have reviewed (and updated if needed) the documentation regarding this change

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: api/datastorage-api/src/main/java/com/cognifide/aet/vs/MetadataDAO.java [hunk 1]
```diff
@@ -100,4 +100,10 @@
    * @return list of projects in as DBKeys
    */
   Collection<DBKey> getProjects(String company) throws StorageException;
+
+  /**
+   * @param dbKey - key with project and company name
+   * @return true when database for given dbKey exists
```
Comment (tkaik) [inline api/datastorage-api/src/main/java/com/cognifide/aet/vs/MetadataDAO.java]: It should probably return `true` also when the database does not exist but the `allowAutoCreate` flag is set to true :) Please update javadoc

File: api/communication-api/src/main/java/com/cognifide/aet/communication/api/execution/ProcessingStatus.java [hunk 2]
```diff
@@ -35,6 +35,11 @@
    */
   FATAL_ERROR,
 
+  /**
+   * An error occured when accessing database.
+   */
+  PROCESSING_ERROR,
```
Comment (tkaik) [inline api/communication-api/src/main/java/com/cognifide/aet/communication/api/execution/ProcessingStatus.java]: If this status is only returned when the database does not exist (and can't be created), please rename it to e.g. `MISSING_DATABASE`

**Comment by plutasnyy** [comment]
Please update a changelog ;)

**Comment by bambroziak** [comment]
Tested and accepted - I've disabled automatic creation of database on AET MongoDB Client in Karaf.
Then I've runned AET suite with not existing company and project by aet.sh script on server.
Database was not created and test suite didin't get executed.

But here the issue I've spotted, there is no any "user-friendly" output, there was only timeout.
Probably there should be information for user that database could not be executed in output logs.

QA Covered.

**Comment by bambroziak** [comment]
Tested and accepted.
I've disabled automatic creation of database on AET MongoDB Client in Karaf.
Then I've rebuilded Integration env with branch changes.
Next I've runned AET suite with not existing company and project by aet.sh script on server.
Database was not created and test suite didin't get executed.

"No database found for company: 'companyname', project: 'projectname'." - result is displaying, which is expected. Without changes only timeout display.

QA Covered.

**Pull request merged**